src/extensions/util/dmenu.rs

```diff
@@ -91,32 +91,34 @@ impl DMenuConfig {
             ..
         } = self;
 
-        let mut s = format!(
-            "-nb {} -nf {} -sb {} -m {}",
+        let mut flags = vec![
+            "-nb".to_string(),
             bg_color.as_rgb_hex_string(),
+            "-nf".to_string(),
             fg_color.as_rgb_hex_string(),
+            "-sb".to_string(),
             selected_color.as_rgb_hex_string(),
-            screen_index,
-        );
+            "-m".to_string(),
+            screen_index.to_string(),
+        ];
 
         if n_lines > 0 {
-            s.push_str(&format!(" -l {n_lines}"))
+            flags.extend_from_slice(&["-l".to_string(), n_lines.to_string()]);
         }
 
         if password_input {
-            s.push_str(" -P");
+            flags.extend_from_slice(&["-P".to_string()]);
```

Comment:
I really don't understand why this is using `extend_from_slice` rather than `push`